### PR TITLE
Add PR message if branch desc mismatch branch target

### PR DESCRIPTION
**Pull request by alamirault**
Continue work https://github.com/symfony-tools/carsonbot/pull/198 begin by @94noni which fix https://github.com/symfony-tools/carsonbot/issues/176

I applied suggestions and add phpunit tests.

In order to add integration test https://github.com/symfony-tools/carsonbot/blob/master/tests/Controller/WebhookControllerTest.php#L59. I think you have to create new branch `Nyholm-patch-11`, right ?

### CODE REVIEW

**Review by weaverryan — COMMENTED** [review]
Thanks for this - left one comment!

File: src/Subscriber/MismatchBranchDescriptionSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        // @see symfony/symfony/.github/PULL_REQUEST_TEMPLATE.md
+        if (!$s->containsAny('Branch?')) {
+            return null;
+        }
+
+        $descriptionBranch = $s->match('/.*Branch.*/');
+
+        $descriptionBranch = $descriptionBranch[0]; // row matching
+        $descriptionBranch = \explode('|', $descriptionBranch)[2]; // get the version
```
[the remark this answers is not in the document]
Comment (alamirault) [reply src/Subscriber/MismatchBranchDescriptionSubscriber.php]: Check + test added

File: src/Subscriber/MismatchBranchDescriptionSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        $targetBranch = $data['pull_request']['base']['ref'];
+        if ($targetBranch === $descriptionBranch) {
+            return;
+        }
+
+        $this->issueApi->commentOnIssue($event->getRepository(), $number, <<<TXT
+Hey!
+
+Thanks for your PR. You are targeting branch "$targetBranch" but it seems your PR description refers to branch "$descriptionBranch".
+Could you update the PR description please or maybe the target branch is wrong? This helps core maintainers a lot.
```
Comment (ogizanagi) [inline src/Subscriber/MismatchBranchDescriptionSubscriber.php:51]: This reads a bit weird to me.


```suggestion
Please update the PR description. Or maybe the target branch is wrong? This helps core maintainers a lot.
```

or

```suggestion
Could you please update the PR description or maybe the target branch is wrong? This helps core maintainers a lot.
```

or

```suggestion
Could you update the PR description, please ? Or maybe the target branch is wrong? This helps core maintainers a lot.
```

?

File: src/Subscriber/MismatchBranchDescriptionSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+use App\Api\Issue\IssueApi;
+use App\Event\GitHubEvent;
+use App\GitHubEvents;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\String\UnicodeString;
+
+/**
+ * @author Antoine Makdessi <amakdessi@me.com>
```
Comment (94noni) [inline src/Subscriber/MismatchBranchDescriptionSubscriber.php:13]: I will be your coauthor with great pleasure but please add your email also as taking care on continuation!

File: src/Subscriber/MismatchBranchDescriptionSubscriber.php (hunk 2 above)
Comment (alamirault) [reply src/Subscriber/MismatchBranchDescriptionSubscriber.php]: Changed to `Could you update the PR description or change target branch? This helps core maintainers a lot.`

File: src/Subscriber/MismatchBranchDescriptionSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        $targetBranch = $data['pull_request']['base']['ref'];
+        if ($targetBranch === $descriptionBranch) {
+            return;
+        }
+
+        $this->issueApi->commentOnIssue($event->getRepository(), $number, <<<TXT
+Hey!
+
+Thanks for your PR. You are targeting branch "$targetBranch" but it seems your PR description refers to branch "$descriptionBranch".
+Could you update the PR description or change target branch? This helps core maintainers a lot.
```
Comment (94noni) [inline src/Subscriber/MismatchBranchDescriptionSubscriber.php:52]: This helps core maintainers a lot to triage PRs regarding if they are features or bugs

**Review by 94noni — APPROVED** [review]

**Pull request merged**